earth_enterprise/src/installer/install_fusion.sh

```diff
@@ -822,10 +822,10 @@ install_or_upgrade_asset_root()
         if [ $IS_SLAVE == false ]; then
             # TODO: Verify this logic -- this is what is defined in the installer documentation, but need confirmation
             if [ $NEW_GEGROUP == true ] || [ $NEW_GEFUSIONUSER == true ]; then
-                NOCHOWN=""
-                UPGRADE_MESSAGE="\nThe upgrade will fix permissions for the asset root and source volume. This may take a while.\n"
+                UPGRADE_MESSAGE="WARNING: The installer detected the asset root may have \
```

Comment:
There's a potential issue on line 818 above (I can't put the comment there so I'm putting it here). If you do the following:

1. Install Fusion
2. Put a bunch of files with the wrong ownership under /gevol/fusion
3. Uninstall Fusion
4. Install Fusion again

then this script will run a potentially costly chown operation. The key is that you uninstall Fusion and install it again rather than upgrade in-place. In that case it will be considered a "new" installation even though /gevol/assets already exists and it will run the chmod.

I'm not sure how much this matters. It's likely to be a rare occurrence, so I could be convinced not to worry about it.

Reply:
Another point here - this message only prints out if the script has to create gefusionuser or gegroup. If the user and group already exist we'll never see this message. Similarly, if the user and group don't exist but the permissions are correct on /gevol/assets, this message will print out anyway.

The logic in this area is kind of convoluted and could probably use a good refactor.